packages/grid/x-data-grid/src/hooks/utils/useTimeout.ts

```diff
@@ -0,0 +1,34 @@
+import { useLazyRef } from './useLazyRef';
+import { useOnMount } from './useOnMount';
+
+class Timeout {
+  static create() {
+    return new Timeout();
+  }
+
+  currentId: number = 0;
+
+  start(delay: number, fn: Function) {
+    this.clear();
+    this.currentId = setTimeout(fn, delay);
+  }
+
+  clear = () => {
+    if (this.currentId !== 0) {
+      clearTimeout(this.currentId);
+      this.currentId = 0;
+    }
```

Comment:
Would it be equality fast but less JavaScript code with?

```suggestion
   clearTimeout(this.currentId);
```

Reply:
It's a small cost but in some places it matters.
Details here: https://github.com/mui/material-ui/pull/37512

I have a few of these utils that should be migrated to core at some point.

Reply:
> It's a small cost but in some places it matters. Details here: https://github.com/mui/material-ui/pull/37512

https://github.com/quasarframework/quasar/issues/15203 seem to have faced something similar. 

> I have a few of these utils that should be migrated to core at some point.

`useLazyRef`: it could be great to comment on it to link https://github.com/facebook/react/issues/14490 I guess it's why it's here. Useful when doing something a bit heavy.